src/Microsoft.AspNet.OData.Shared/Formatter/Deserialization/ODataResourceDeserializer.cs

```diff
@@ -314,10 +318,43 @@ public virtual object CreateResourceInstance(IEdmStructuredTypeReference structu
                     }
                 }
 
-                ODataEntityReferenceLinkBase entityReferenceLink = childItem as ODataEntityReferenceLinkBase;
-                if (entityReferenceLink != null)
+                ODataEntityReferenceLinksWrapper referenceLinksWrapper = childItem as ODataEntityReferenceLinksWrapper;
+                if (referenceLinksWrapper != null)
                 {
-                    // ignore entity reference links.
+                    IEdmCollectionTypeReference edmCollectionTypeReference = edmProperty.Type.AsCollection();
+                    ODataResourceSetWrapper rSetWrapper = CreateResourceSetWrapper(edmCollectionTypeReference, referenceLinksWrapper.EntityReferenceLinks, readContext);
+                    if (rSetWrapper != null)
+                    {
+                        if (edmProperty == null)
+                        {
+                            ApplyDynamicResourceSetInNestedProperty(resourceInfoWrapper.NestedResourceInfo.Name,
+                                resource, structuredType, rSetWrapper, readContext);
+                        }
```

Comment:
dynamic reference link? I think we should exclude dynamic property reference link

Reply:
@xuzhg is there any reason for that?